### PR TITLE
Add go 1.17 to the build matrix

### PR DESCRIPTION
**Pull request by kishaningithub**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [1.15.x, 1.16.x]
+        go-version: [1.15.x, 1.16.x, 1.17.x]
```
Comment (vmihailenco) [inline .github/workflows/build.yml:15]: Thanks :+1:  Could you also remove 1.15.x?

Comment (kishaningithub) [reply .github/workflows/build.yml]: Done

**Comment by kishaningithub** [comment]
@vmihailenco Should it be identitical to what we have for go-redis? Also, Do we want golang-cli lint as a separate pipeline or can it be a step within the same pipeline

**Comment by vmihailenco** [comment]
>@vmihailenco Should it be identitical to what we have for go-redis?

I assume this question is about go-redis/rate. Yes, something similar would be fine.

>Also, Do we want golang-cli lint as a separate pipeline or can it be a step within the same pipeline

Separate pipelines can be run in parallel which is nice. Also it is simple enough so why not?

**Comment by vmihailenco** [comment]
Build is passing - merging and thanks.

**Pull request merged**